### PR TITLE
Servlet 6.0 Response setCharacterEncoding

### PR DESCRIPTION
**Pull request by pmd1nh**
https://github.com/openliberty/open-liberty/issues/20423

Clarify the behaviour of response : setCharacterEncoding(null), setContentType(null), setLocale(null)

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
Added more tests

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat

**Comment by wraschke** [comment]
Run #libby

**Comment by pmd1nh** [comment]
rebased + updated to .classpath

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat

**Review by isaacrivriv — DISMISSED** [review]
I think these changes look good. Thanks for working on this!

File: dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTServletResponse.java [hunk 1]
```diff
@@ -2132,9 +2140,12 @@ private boolean isCharEncodingSet()
                 return isCharEncodingExplicit;
             }
         }
+        // if contentType is null, it means no setContentType has called before but setCharacterEncoding may have called
+        // so check isCharEncodingExplicit and return;  Not sure why only return if ContentType is set ?
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTServletResponse.java]: `Not sure why only return if ContentType is set?` Is this still an outstanding question we need to answer? Can we clean up the comment?

File: dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTServletResponse.java [hunk 2]
```diff
@@ -2132,9 +2140,12 @@ private boolean isCharEncodingSet()
                 return isCharEncodingExplicit;
             }
         }
+        // if contentType is null, it means no setContentType has called before but setCharacterEncoding may have called
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTServletResponse.java]: `was called` vs `has called`

Comment (pnicolucci) [reply dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTServletResponse.java]: `may have been called`

File: dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/srt/SRTServletResponse60.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    }
+
+    @Override
+    public void initForNextResponse(IResponse resp) {
+        super.initForNextResponse(resp);
+    }
+
+    @Override
+    public void setCharacterEncoding(String encoding) {
+        if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/srt/SRTServletResponse60.java:41]: General question for all of these methods is, basically they are copied from the 40 class, using existing messages correct? Then some new updates for Servlet 6.0, correct?

Comment (pmd1nh) [reply dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/srt/SRTServletResponse60.java:41]: yes, it extends the 4.0.

**Comment by pmd1nh** [comment]
updated comments

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat

**Comment by pmd1nh** [comment]
Updated a path where getContentType() would trigger setContentType(null) automatically when there is no header's content-type.  Header's no contentype should return without setContentType(null) cuz that will clear out the set encoding.

#build

**Comment by pmd1nh** [comment]
Some errors in the microprofile.jwt.  the output.txt shows lot of error/warning
```
[10/19/2022 02:05:49:586 PDT] 001 LibertyServer                  checkLogsForErrorsAndWarnings  I Error/warning found in log /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/jwt21tckAudEnv/logs/messages.log: [10/19/22, 2:03:42:459 PDT] 0000002c com.ibm.ws.security.mp.jwt.tai.MicroProfileJwtTAI            E CWWKS5523E: The MicroProfile JWT feature cannot authenticate the request because the token that is included in the request cannot be validated. JWT_TOKEN_AGE_AFTER_CURRENT_TIME
[10/19/2022 02:05:49:587 PDT] 001 LibertyServer                  checkLogsForErrorsAndWarnings  I Error/warning found in log /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/jwt21tckAudEnv/logs/messages.log: [10/19/22, 2:03:42:807 PDT] 0000002c com.ibm.ws.security.mp.jwt.tai.MicroProfileJwtTAI            E CWWKS5523E: The MicroProfile JWT feature cannot authenticate the request because the token that is included in the request cannot be validated. JWT_TOKEN_AGE_AFTER_CURRENT_TIME
[10/19/2022 02:05:49:587 PDT] 001 LibertyServer                  checkLogsForErrorsAndWarnings  I Error/warning found in log /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/jwt21tckAudEnv/logs/messages.log: [10/19/22, 2:03:42:948 PDT] 00000038 com.ibm.ws.security.mp.jwt.tai.MicroProfileJwtTAI            E CWWKS5523E: The MicroProfile JWT feature cannot authenticate the request because the token that is included in the request cannot be validated. JWT_TOKEN_AGE_AFTER_CURRENT_TIME
[10/19/2022 02:05:49:587 PDT] 001 LibertyServer                  checkLogsForErrorsAndWarnings  I Error/warning found in log /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/jwt21tckAudEnv/logs/messages.log: [10/19/22, 2:03:43:011 PDT] 00000038 com.ibm.ws.security.mp.jwt.tai.MicroProfileJwtTAI            E CWWKS5523E: The MicroProfile JWT feature cannot authenticate the request because the token that is included in the request cannot be validated. JWT_TOKEN_AGE_AFTER_CURRENT_TIME
```

**Comment by pmd1nh** [comment]
#libby

**Review by pnicolucci — APPROVED** [review]

**Comment by pnicolucci** [comment]
#libby

**Pull request merged**